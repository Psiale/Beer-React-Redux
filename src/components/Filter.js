import React from 'react';
import PropTypes from 'prop-types';
import styles from './Filter.module.css';

const categories = [
  'All',
  'Pale Ale',
  'Pale',
  'Lager',
  'Porter',
  'Wheat',
  'Stout',
  'IPA',
  'IMP',
];

const CategoryFilter = ({ handleFilterChange }) => (
  <div className={styles.header}>
    <select
      className={styles.select}
      onChange={handleFilterChange}
      name="bookCategories"
      id="booksCategories"
    >
      {categories.map(category => (
        <option className={styles.option} key={category}>{category}</option>
      ))}
    </select>
  </div>
);

CategoryFilter.propTypes = {
  handleFilterChange: PropTypes.func.isRequired,
};

export default CategoryFilter;
