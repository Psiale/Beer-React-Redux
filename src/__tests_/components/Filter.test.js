import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Filter from '../../components/Filter';

configure({ adapter: new Adapter() });

const component = <Filter handleFilterChange={() => {}} />;
it('renders correctly', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('all the divs are properly render', () => {
  const wrapper = shallow(<Filter handleFilterChange={() => {}} />);
  const divs = wrapper.find('div');
  expect(divs).toHaveLength(1);
});

it('all the select are properly render', () => {
  const wrapper = shallow(<Filter handleFilterChange={() => {}} />);
  const divs = wrapper.find('select');
  expect(divs).toHaveLength(1);
});
