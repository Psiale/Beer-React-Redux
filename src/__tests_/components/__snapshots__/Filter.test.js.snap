// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header"
>
  <select
    className="select"
    id="booksCategories"
    name="bookCategories"
    onChange={[Function]}
  >
    <option
      className="option"
    >
      All
    </option>
    <option
      className="option"
    >
      Pale Ale
    </option>
    <option
      className="option"
    >
      Pale
    </option>
    <option
      className="option"
    >
      Lager
    </option>
    <option
      className="option"
    >
      Porter
    </option>
    <option
      className="option"
    >
      Wheat
    </option>
    <option
      className="option"
    >
      Stout
    </option>
    <option
      className="option"
    >
      IPA
    </option>
    <option
      className="option"
    >
      IMP
    </option>
  </select>
</div>
`;
